src/autopas/containers/hierarchicalGrids/HierarchicalGrids.h

```diff
@@ -563,77 +538,124 @@ class HierarchicalGrids : public ParticleContainerInterface<Particle> {
 
-  private:
+    return {level, cellID};
+  }
 
   /**
-   * @brief Get the number Of cross-levels of the H-Grid
-   * 
-   * @param levels 
-   * @return unsigned int 
+   * Calculate the number of cross-level interactions for the present number of hierarchy levels
+   *
+   * @return size_t number of cross-level interactions
    */
-  unsigned int getNumberOfCrossLevels(unsigned int levels) {
+  size_t getNumberOfCrossLevelInteractions() const {
+    size_t numberOfCrossLevels = 0;
 
-    unsigned int numberOfCrossLevels = 0;
-    for (unsigned int i = 1; i < levels; i++)
-    {
-      numberOfCrossLevels += i;
+    for (unsigned int largerLevel = 0; largerLevel < _hierarchyLevels.size(); largerLevel++) {
+      for (unsigned int smallerLevel = largerLevel + 1; smallerLevel < _hierarchyLevels.size(); smallerLevel++) {
+        numberOfCrossLevels++;
+      }
     }
-    
+
     return numberOfCrossLevels;
   }
 
   /**
-  * @brief Vector containing the HGrid's different hierarchy levels of Linked Cells.
-  * 
-  * @note This can not be a std::vector<LinkedCells<Particle>> because there is no copy constructor for LinkedCells. 
-  */
+   * Set the cross-level interaction pairs, i.e. the which larger level is checked with which smaller level
+   *
+   */
+  void setCrossLevelInteractionPairs() {
+    for (unsigned int largerLevel = 0; largerLevel < _hierarchyLevels.size(); largerLevel++) {
+      for (unsigned int smallerLevel = largerLevel + 1; smallerLevel < _hierarchyLevels.size(); smallerLevel++) {
+        _crossLevelInteractions.emplace_back(largerLevel, smallerLevel);
+      }
+    }
+  }
+
+  /**
+   * @brief Vector containing the HGrid's different hierarchy levels of Linked Cells.
+   *
+   * @note This can not be a std::vector<LinkedCells<Particle>> because there is no copy constructor for LinkedCells.
+   */
   std::deque<LinkedCells<Particle>> _hierarchyLevels;
 
-  std::array<double, 3> _boxMin; 
-  std::array<double, 3> _boxMax; 
+  std::vector<std::tuple<size_t, size_t>> _crossLevelInteractions;
+
+  std::array<double, 3> _boxMin;
+  std::array<double, 3> _boxMax;
   double _cutoff;
-  double _skinPerTimestep; 
+  double _skinPerTimestep;
   unsigned int _rebuildFrequency;
   double _cellSizeFactor;
   autopas::LoadEstimatorOption _loadEstimator;
 
   /**
-   * @brief Vector containing the upper boundaries of the grid levels. 
-   * 
+   * @brief Vector containing the upper boundaries of the grid levels.
+   *
    * For example, the first element describes the upper boundary of the largest H-Grid level,
    * i.e. the largest particles fitting in the topmost level.
    * By design '_hierarchicalGridBorders' should have the same number of levels as the H-Grid.
-   * 
+   *
    */
   std::vector<double> _hierarchicalGridBorders;
 
+  /**
+   * Level offset required for encoding and decoding the current HG-level and the cellIndex
+   * @see HierarchicalGrids::getParticleImpl.
+   *
+   */
+  const size_t _levelOffset = 1e15;
```

Comment:
```suggestion
  static constexpr size_t _levelOffset = 1e15;
```
`const` members are discouraged because they make objects non-movable

Reply:
Done